### ISSUE TITLE
many boxes in one build

### ISSUE THREAD

**Issue by Jarema** [wercker/support#80]
I have a situation where i need to use golang box to compile go code, and then node box, to install all npm dependencies and run gulp/grunt tasks.

A walkaround i see is to create new box with both node and golang installed, but it feels not optimal.
Other solution is to use golang box for build and node box for deploy to build everything that it's needed, but i'm still not sure if it's proper way, as some build tasks would be done in deploy phase.

Best solution would be to be able to have more boxes per build, or being able to have two  build phases - one with golang an one with node box. Is it possible?

**Comment by mies** [comment]
Supporting multiple build phases is definitely something we're interested in supporting. I am curious though to hear if this concerns multiple applications, i.e a frontend and a backend. If so it could be an idea to split up your app?

**Comment by Jarema** [comment]
Yes, usually it's possible to split app if it's not split by design to frontend/backend etc, but sometimes You end up having small fullstack app and dividing it is simply an overengineering, or You might have app that uses few technologies by design (like isomorphic react app with node for rendering react classes and golang as API server, aaand js for frontend). That would need one box with both node and golang which is not optimal. In such case -  two build phases would be much better solution.
